### ISSUE TITLE
RPC response crash for debug_traceTransaction due to stringify

### ISSUE THREAD

**Issue by Archethect** [NomicFoundation/hardhat#3272]
Context:

I am setting up a local hardhat node in combination with blockscout to do some local tests.
When blockscout starts fetching data through the RPC endpoint, the hardhat node fails with the following error:

```
/usr/src/app/node_modules/hardhat/src/internal/hardhat-network/jsonrpc/handler.ts:155
    res.end(JSON.stringify(rpcResp));
                 ^
RangeError: Invalid string length
    at JSON.stringify (<anonymous>)
    at JsonRpcHandler._sendResponse (/usr/src/app/node_modules/hardhat/src/internal/hardhat-network/jsonrpc/handler.ts:155:18)
    at Server.JsonRpcHandler.handleHttp (/usr/src/app/node_modules/hardhat/src/internal/hardhat-network/jsonrpc/handler.ts:48:12)
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
```

In my case the calls that brake it are:
```
{"jsonrpc":"2.0", "method":"debug_traceTransaction", "params":["0xe751022dd3074ae061939331ceca6b05e2f679c9f92aff6a9e9ae9a033b6fc84", {"tracer": "callTracer"}], "id":1}
```

Hardhat version: v2.10.1
Node version: v14

I suppose the issue is related to https://github.com/NomicFoundation/hardhat/pull/2512 and https://github.com/NomicFoundation/edr/issues/392

**Comment by hadfieldn** [comment]
Running into this issue as well.

**Comment by fvictorio** [comment]
Thanks for reporting this @Archethect, I can confirm this is a bug.

**Comment by srinjoychakravarty** [comment]
Running into this exact same issue as of yesterday:

<img width="964" alt="Screenshot 2022-11-17 at 5 09 36 AM" src="https://user-images.githubusercontent.com/1481037/202420121-646952c1-b594-4b28-a64c-e37384171f53.png">


>     "hardhat": "^2.12.2",
package.json

**Comment by SuperHero123123** [comment]
> Thanks for reporting this @Archethect, I can confirm this is a bug.

meet the same bug today, is there anyway to resolve this?

**Comment by unparalleled-js** [comment]
I noticed you (or blockscout) is using `"tracer": "callTracer"`, but when looking at the code and the docs, I don't see that as an option anywhere. I would like to use the call tracer as well but can't seem to get it to work.

**Comment by antazoey** [comment]
I am getting this error, and in my response, it is mostly empty zeroes... A lot of empty byte strings. I tried on Anvil, Geth --dev, eth Tester... They have the expected data response size. So maybe something is messing up there?

**Comment by kanej** [comment]
This has been raised in the EDR repository as well: https://github.com/NomicFoundation/edr/issues/543

**Comment by kanej** [comment]
I am closing this issue as we have released the EDR fix for stringifying the `debug_traceTransaction` response:

https://github.com/NomicFoundation/hardhat/releases/tag/hardhat%402.22.8

**Comment by schaable** [comment]
Released in [Hardhat v2.22.11](https://github.com/NomicFoundation/hardhat/releases/tag/hardhat%402.22.11).
